CardsCashCasino/Data/HoldEmUtil.cs

```diff
@@ -0,0 +1,335 @@
+﻿/*
+ *  Module Name:HoldEmUtil.cs
+ *  Purpose: Contains static utility methods used to determine the winner of a game of Texas Hold 'Em.
+ *  Inputs: The set of community cards and a player's hole cards.
+ *  Outputs: The ranking, optimal 5-card hand, and the value used to break ties.
+ *  Additional code sources: None
+ *  Developers: Ethan Berkley
+ *  Date: 11/08/2024
+ *  Last Modified: 11/08/2024
+ *  Preconditions: Hole card lists are of length 2, Community card lists are of length >= 3, the input to the tiebreaker function was returned by the GetScore function
+ *  Postconditions: None
+ *  Error/Exception conditions: Only if Preconditions were violated.
+ *  Side effects: None
+ *  Invariants: None
+ *  Known Faults: None encountered
+ */
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CardsCashCasino.Data
+{
+    /// <summary>
+    /// Utility class that will be used to determine a hand's ranking and tie breaking score, which is used at the conclusion of a round.
+    /// Defines an enum and two functions:
+    /// The enum, Ranking, represents the Rank of the poker hand. i.e. Royal Flush, Four of a Kind, Full House, etc.
+    /// The first function, GetScore, takes in the community cards and the player's hole cards, and returns the optimal hand and it's ranking.
+    /// The second function, KickerValue, takes in a 5-card hand returned by GetScore, 
+    ///     and returns the value used to break ties between hands of the same rank.
+    /// </summary>
+    public class HoldEmUtil
+    {
+        /// <summary>
+        /// Enum representing the rank of a 5-card poker hand.
+        /// </summary>
+        public enum Ranking
+        {
+            ROYAL_FLUSH,
+            STRAIGHT_FLUSH,
+            FOUR_OF_A_KIND,
+            FULL_HOUSE,
+            FLUSH,
+            STRAIGHT,
+            THREE_OF_A_KIND,
+            TWO_PAIR,
+            PAIR,
+            HIGH_CARD
+        }
+
+        
+        
+
+        /// <summary>
+        /// Given the current 5-card community card pool, and that the current hand has exactly 2 cards, 
+        /// create and return the best possible hand and ranking.
+        /// 
+        /// In the event of a tie in rankings, the tie would be broken by calling KickerValue on each of the hands. 
+        /// If the two return values are the same, split the pot.
+        /// </summary>
+        /// <param name="community">The 5-card list of community cards</param>
+        /// <param name="cards">The 2-card list representing player's hole cards.</param>
+        /// <returns>A 5-card list representing the optimal hand, and the ranking it would give.</returns>
+        public static Tuple<List<Card>, Ranking> GetScore(List<Card> community, List<Card> cards)
```

Comment:
Add some checks at the start to ensure the Lists passed in have the correct sizes.
I think that we could allow for the community list to pass in 3 cards, then we can get the current best hand the player has at every point in the game. Get3CardPermutations presumably would still be fine on a list of 3 items.